workspace/exercise.go

```diff
@@ -59,3 +65,59 @@ func (e Exercise) HasMetadata() (bool, error) {
 	}
 	return false, err
 }
+
+// HasLegacyMetadata checks for the presence of a legacy exercise metadata file.
+// If there is no such file, it could also be an unrelated directory.
+func (e Exercise) HasLegacyMetadata() (bool, error) {
+	_, err := os.Lstat(e.LegacyMetadataFilepath())
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	if err == nil {
+		return true, nil
+	}
+	return false, err
+}
+
+// MigrationStatus represents the result of migrating a legacy metadata file.
+type MigrationStatus int
+
+// MigrationStatus
+const (
+	MigrationStatusNoop MigrationStatus = iota
+	MigrationStatusMigrated
+	MigrationStatusRemoved
+)
+
+func (m MigrationStatus) String(e Exercise) string {
```

Comment:
Is it awkward that this takes an `Exercise`. I'm not sure if there's a better way to do this.

Reply:
>  Making it a method on Exercise doesn't seem right.

Why is that? I actually think that seems fine, but I fear that I'm missing something.

Reply:
No, let's not include the metadata location.